python/shotgun_desktop/upgrade_startup.py

```diff
@@ -48,6 +48,9 @@ def upgrade_startup(splash, sgtk, app_bootstrap):
     if not _supports_get_from_location_and_paths(sgtk):
         return False
 
+    if 'SGTK_DESKTOP_DISABLE_UPDATE' in os.environ:
```

Comment:
To clarify: it disables auto-updates to the desktop startup code. In no way does this disable updates to the core and pipeline configuration internally used to run the Shotgun Desktop. This can be done my following the steps documented [here](https://support.shotgunsoftware.com/hc/en-us/articles/219040428-How-Can-I-Setup-a-Shared-Studio-Site-Config-).